platform/docs/docs/platform/services/ui/ui-customization-service.md

```diff
@@ -0,0 +1,196 @@
+Note that mode specific customizations are CLEARED before this call is made
+by a call to the `uiCustomizationService.onModeEnter` from the `ExtensionManager`.
+
+The following example shows first the registration of the default values,
+in this case they are base types which are inheritted later:
+
+```js
+// In the cornerstone extension  getCustomizationModule:
+const getCustomizationModule = () => ([
+  {
+    name: 'default',
+    value: [
+      {
+        id: 'ohif.cornerstoneContextMenu',
+        commands: [... default commands for the context menu],
+        content: CornerstoneOverlay,
+        contentProps: ....
+      },
+      {
+        id: 'ohif.menuItem',
+        content: CornerstoneMenuItem,
+      },
+    ],
+  },
+]);
+```
+
+Then, in the configuration file one might have a custom overlay definition:
+
+```js
+// in the APP_CONFIG file set the top right area to show the patient name
+// using PN: as a prefix when the study has a non-empty patient name.
+whiteLabeling: {
+  cornerstoneOverlayTopRight: {
+    id: 'cornerstoneOverlayTopRight',
+    items: [
+      {
+        id: 'PatientNameOverlay',
+        // Note the overlayItem as a parent type - this provides the
+        // rendering functionality to read the attribute and use the label.
+        customizationType: 'ohif.overlayItem',
+        attribute: 'PatientName',
+        label: 'PN:',
+      },
+    ],
+  }
+},
+```
+
+In the mode customization, the context menu is then customized:
+
+```js
+// First in ExtensionManager, the mode customizations are cleared by calling:
+UICustomizationService.onModeEnter();
+
+// Then in the mode itself, customizations can be registered:
+onModeEnter() {
+  ...
+  uiCustomizationService.addModeCustomizations(
+    {
+      id: 'cornerstoneContextMenu',
+      // Note the type is the previously registered ohif.cornerstoneContextMenu
+      customizationType: 'ohif.cornerstoneContextMenu',
+      items: [
+        // Custom definitions for hte context menu here.
+      ],
+    }
+}
+```
+
+## Mode Customizations
+The mode customizations are retrieved via the `getModeCustomization` function,
+providing an id, and optionally a default value.  The retrieval will return,
+in order:
+
+1. Global customization with the given id.
+2. Mode customization with the id.
+3. The default value specified.
+
+The return value then inherits the inheritted type value, so that the value can
+be typed and have default values and functionality provided.  The object
+can then be used in a way defined by the extension provided that customization
+point.
+
+```ts
+   cornerstoneOverlay = uiConfigurationService.getModeCustomization("cornerstoneOverlay", {customizationType: "ohif.cornerstoneOverlay", ...});
+   const { component: overlayComponent, props} = uiConfigurationService.getComponent(cornerstoneOverlay);
+   return (<defaultComponent {...props} overlay={cornerstoneOverlay}....></defaultComponent>);
+```
```

Comment:
If i'm not mistaken this is kind of behind the scene of UICustomizationService and how it works right? I don't think we need to explain that. Rather we need to show examples on how to configure stuff.

Reply:
It is part and parcel of how one defines customizations for use, so I think it is actually fairly important to define here.